src/Symfony/Component/VarDumper/Cloner/VarCloner.php

```diff
@@ -86,8 +90,13 @@ protected function doClone($var)
             }
             foreach ($vals as $k => $v) {
                 // $v is the original value or a stub object in case of hard references
-                $refs[$k] = $cookie;
-                if ($zvalIsRef = $vals[$k] === $cookie) {
+
+                if (\PHP_VERSION_ID < 70400 || (($zvalIsRef = null !== $r = \ReflectionReference::fromArrayElement($vals, $k)) && (!$zvalIsRef = 2 < $r->getRefcount()) && \is_array($v))) {
```

Comment:
On PHP 7.4, detecting a reference needs mutating the value when it is an array with refcount=1, for catching circular refs.
See https://github.com/php/php-src/pull/4380#issuecomment-513564327